matlab/CMakeLists.txt

```diff
@@ -212,56 +233,150 @@ if(MATLAB_BUILD_TESTS)
     # C++ libraries that are built from source.
     build_arrow(BUILD_GTEST)
   else()
+    # GTest is found, on Windows, IMPORTED_LOCATION needs to be set to indicate where the shared
+    # libraries live
+    if(WIN32)
+      set(GTEST_SHARED_LIB_DIR "${GTEST_ROOT}/bin")
+      set(GTEST_SHARED_LIBRARY_FILENAME
+          "${CMAKE_SHARED_LIBRARY_PREFIX}gtest${CMAKE_SHARED_LIBRARY_SUFFIX}")
+      set(GTEST_SHARED_LIBRARY_LIB
+          "${GTEST_SHARED_LIB_DIR}/${GTEST_SHARED_LIBRARY_FILENAME}")
+
+      set(GTEST_MAIN_SHARED_LIB_DIR "${GTEST_ROOT}/bin")
+      set(GTEST_MAIN_SHARED_LIBRARY_FILENAME
+          "${CMAKE_SHARED_LIBRARY_PREFIX}gtest_main${CMAKE_SHARED_LIBRARY_SUFFIX}")
+      set(GTEST_MAIN_SHARED_LIBRARY_LIB
+          "${GTEST_MAIN_SHARED_LIB_DIR}/${GTEST_MAIN_SHARED_LIBRARY_FILENAME}")
+
+      set_target_properties(GTest::gtest PROPERTIES IMPORTED_LOCATION
+                                                    "${GTEST_SHARED_LIBRARY_LIB}")
+
+      set_target_properties(GTest::gtest_main
+                            PROPERTIES IMPORTED_LOCATION
+                                       "${GTEST_MAIN_SHARED_LIBRARY_LIB}")
+    endif()
+
     find_package(Arrow)
     if(NOT Arrow_FOUND)
       # Trigger an automatic build of the Arrow C++ libraries.
       build_arrow()
     endif()
   endif()
+
+  # On Windows: copy the gtest and gtest_main runtime DLLs into the directory where the
+  # MATLAB C++ tests reside, since Windows require runtime DLLs that are depended on by
+  # executables to be in the same directory (or on the %PATH%).
+  if(WIN32)
+    set(MATLAB_TESTS_DIR "${MATLAB_BUILD_OUTPUT_DIR}")
+
+    get_property(GTEST_SHARED_LIB
+                 TARGET GTest::gtest
+                 PROPERTY IMPORTED_LOCATION)
+
+    get_property(GTEST_MAIN_SHARED_LIB
+                 TARGET GTest::gtest_main
+                 PROPERTY IMPORTED_LOCATION)
+
+    add_custom_target(copy_gtest_to_tests_dir ALL
```

Comment:
I don't try this but is this target always run even when GTest files aren't rebuilt?
Can we use `BYPRODUCTS` and `DEPENDS` to prevent needless execution?

Reply:
This piece of code is not necessary anymore after adding the path of the shared libraries to the test environment path.